lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -1423,11 +1580,21 @@ namespace {
 
     /// Note that the given actor member is isolated.
     /// @param context is allowed to be null if no context is appropriate.
-    void noteIsolatedActorMember(ValueDecl *decl, Expr *context) {
+    void noteIsolatedActorMember(ValueDecl *decl, Expr *context,
+                                 bool isDistributedActor) {
       // FIXME: Make this diagnostic more sensitive to the isolation context
       // of the declaration.
-      if (auto func = dyn_cast<AbstractFunctionDecl>(decl)) {
-        func->diagnose(diag::actor_isolated_sync_func, 
+      if (isDistributedActor) {
+        // TODO: detect isDistributedActor based on the decl, rather than an explicit boolean
```

Comment:
I agree with this TODO. No reason to ask the clients to think about `distributed`.

Reply:
done!